src/modm/platform/adc/stm32/adc_impl.hpp.in

```diff
@@ -253,14 +259,75 @@ modm::platform::Adc{{ id }}::acknowledgeInterruptFlags(const InterruptFlag_t fla
 	ADC{{ per }}->SR = ~flags.value;
 }
 
+uintptr_t
+modm::platform::Adc{{ id }}::getDataRegisterAddress()
+{
+	return (uintptr_t) &(ADC{{ per }}->DR);
+}
+%% if target["family"] in ["f1","f3"]
+void
+modm::platform::Adc{{ id }}::enableRegularConversionExternalTrigger(
+	RegularConversionExternalTrigger regularConversionExternalTrigger)
+{
+	const auto externalTrigger =
+		(static_cast<uint32_t>(regularConversionExternalTrigger) << ADC_CR2_EXTSEL_Pos);
+	ADC{{ per }}->CR2 |= externalTrigger;
+}
+%%else
+void
+modm::platform::Adc{{ id }}::enableRegularConversionExternalTrigger(
+	ExternalTriggerPolarity externalTriggerPolarity,
+	RegularConversionExternalTrigger regularConversionExternalTrigger)
+{
+	const auto polarity =
+		(static_cast<uint32_t>(externalTriggerPolarity) << ADC_CR2_EXTEN_Pos);
+	const auto externalTrigger =
+		(static_cast<uint32_t>(regularConversionExternalTrigger) << ADC_CR2_EXTSEL_Pos);
+	ADC{{ per }}->CR2 |= ( polarity | externalTrigger);
+}
+%%endif
```

Comment:
@victorandrehc I just wanted to merge and discovered a bug. The trigger source bits are set with bit-and are never cleared.

If `enableRegularConversionExternalTrigger()` is called twice the wrong source will be selected. For example, if you set the trigger to `Event2` first and then try to set it to `Event4` you'll get `Event6`.

```suggestion
%% if target["family"] in ["f1","f3"]
void
modm::platform::Adc{{ id }}::enableRegularConversionExternalTrigger(
	RegularConversionExternalTrigger regularConversionExternalTrigger)
{
	const auto externalTrigger =
		(static_cast<uint32_t>(regularConversionExternalTrigger) << ADC_CR2_EXTSEL_Pos);
	const auto mask = ADC_CR2_EXTSEL_Msk;
	ADC{{ per }}->CR2 = (ADC{{ per }}->CR2 & ~mask) | externalTrigger;
}
%%else
void
modm::platform::Adc{{ id }}::enableRegularConversionExternalTrigger(
	ExternalTriggerPolarity externalTriggerPolarity,
	RegularConversionExternalTrigger regularConversionExternalTrigger)
{
	const auto polarity =
		(static_cast<uint32_t>(externalTriggerPolarity) << ADC_CR2_EXTEN_Pos);
	const auto externalTrigger =
		(static_cast<uint32_t>(regularConversionExternalTrigger) << ADC_CR2_EXTSEL_Pos);
	const auto mask = ADC_CR2_EXTEN_Msk | ADC_CR2_EXTSEL_Msk;
	ADC{{ per }}->CR2 = (ADC{{ per }}->CR2 & ~mask) | polarity | externalTrigger;
}
%%endif
```

Reply:
@victorandrehc Could you please double-check my change?

Reply:
Good eye, it seems fine now. This way it is safe to change the triggers in runtime. Thanks for that.